### PR TITLE
Add fallback when could not retrieve user CY-1876

### PR DESCRIPTION
**Pull request by carlospinho04**

### CODE REVIEW

File: src/main/scala/com/codacy/client/stash/service/UserServices.scala [hunk 1]
```diff
@@ -72,4 +75,23 @@ class UserServices(client: StashClient) {
     client.delete(url)()
   }
 
+  /**
+    *
+    * We need this fallback for the cases when a username contains special characters because the
```
Comment (rtfpessoa) [inline src/main/scala/com/codacy/client/stash/service/UserServices.scala:80]: This is not true right?
The problem is that sometimes you need to find the user by the email and not the username, right?

File: src/main/scala/com/codacy/client/stash/service/UserServices.scala [hunk 2]
```diff
@@ -72,4 +75,23 @@ class UserServices(client: StashClient) {
...
+    */
+  private def getUserFallback(
+      username: String,
+      params: Map[String, String],
+      userResponse: RequestResponse[User]
+  ): RequestResponse[User] = {
+    userResponse.value
+      .orElse {
+        // Since we are using a filter as query parameter this can bring more users and we need to find the one using the username passed by parameter as name
+        getUsers(Option(username), params).value.flatMap(_.find(_.name == username))
```
Comment (rtfpessoa) [inline src/main/scala/com/codacy/client/stash/service/UserServices.scala:92]: Is name the parameter you want?

File: src/main/scala/com/codacy/client/stash/service/UserServices.scala (hunk 1 above)
Comment (carlospinho04) [reply src/main/scala/com/codacy/client/stash/service/UserServices.scala:80]: Nop, the problem is that a user can have an email on the name or other case that makes a name having special characters

File: src/main/scala/com/codacy/client/stash/service/UserServices.scala (hunk 2 above)
Comment (carlospinho04) [reply src/main/scala/com/codacy/client/stash/service/UserServices.scala:92]: Yes, the username that we store is actually the slug so in this case the username will be "test_codacy.com" and the name will be "test@codacy.com"

File: src/main/scala/com/codacy/client/stash/service/UserServices.scala (hunk 1 above)
Comment (rtfpessoa) [reply src/main/scala/com/codacy/client/stash/service/UserServices.scala:80]: Hmmm, where did you get that information from?

All the information I collected on the call was that `/plugins/servlet/applinks/whoami` returns sometimes the email other times the slug.
But if you request the user is always returns the correct things in the correct fields.

**Pull request merged**